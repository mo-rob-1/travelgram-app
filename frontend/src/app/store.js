import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../features/auth/authSlice";
import imageReducer from "../features/images/imageSlice";

export const store = configureStore({
  reducer: {
    auth: authReducer,
    images: imageReducer,
  },
});
