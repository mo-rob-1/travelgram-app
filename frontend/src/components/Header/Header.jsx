import { Link, useNavigate } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { logout, reset } from "../../features/auth/authSlice";
import {
  PageHeader,
  HeaderContainer,
  Nav,
  NavWrapper,
  LogoWrapper,
  List,
  StyledLogoLink,
  StyledNavLink,
} from "./Header.styled";

function Header() {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const { user } = useSelector((state) => state.auth);

  const onLogout = () => {
    dispatch(logout());
    dispatch(reset());
    navigate("/");
  };

  return (
    <PageHeader>
      <LogoWrapper>
        <StyledLogoLink to="/">travelGram</StyledLogoLink>
      </LogoWrapper>
      <NavWrapper>
        <Nav>
          <List>
            {user ? (
              <>
                <li>
                  <StyledNavLink to="/">Home</StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/about">About</StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/our-users">Our Users</StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/upload-image">Upload Image</StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/my-images">My Images</StyledNavLink>
                </li>
                {/* <li>
                  <button>
                    <Link to="/update-profile">Update Profile</Link>
                  </button>
                </li> */}
                <li>
                  <button onClick={onLogout}>Logout</button>
                </li>
              </>
            ) : (
              <>
                <li>
                  <StyledNavLink to="/">Home</StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/about">About</StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/our-users" data-testid="our-users-link">
                    Our Users
                  </StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/login" data-testid="login-link">
                    Login
                  </StyledNavLink>
                </li>
                <li>
                  <StyledNavLink to="/register" data-testid="register-link" style={{ marginRight: 0 }}>
                    Register
                  </StyledNavLink>
                </li>
              </>
            )}
          </List>
        </Nav>
      </NavWrapper>
    </PageHeader>
  );
}

export default Header;
