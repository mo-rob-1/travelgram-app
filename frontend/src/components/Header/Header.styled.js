import styled from "styled-components";
import { Link } from "react-router-dom";

export const PageHeader = styled.header`
  padding-top: 2rem;
  padding-bottom: 2rem;
  @media (min-width: 768px) {
    display: flex;
    justify-content: space-between;
  }
`;

export const List = styled.ul`
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const StyledLogoLink = styled(Link)`
  color: #222;
  text-decoration: none;
  font-size: 1.8rem;
  font-weight: bold;
`;

export const LogoWrapper = styled.div`
  text-align: center;
`;

export const Nav = styled.nav`
  margin-top: 1rem;
  @media (min-width: 768px) {
    margin-top: 0;
  }
`;

export const NavWrapper = styled.div`
  @media (min-width: 768px) {
    display: flex;
    align-items: center;
  }
`;

export const StyledNavLink = styled(Link)`
  color: #222;
  text-decoration: none;
  margin-right: 1rem;
  @media (min-width: 768px) {
    margin-right: 2.5rem;
  }
`;

export const LogoutButton = styled.button`
  cursor: pointer;
  background-color: inherit;
  border: none;
  color: #222;
`;

export const Span = styled.span`
  font-weight: 400;
`;
