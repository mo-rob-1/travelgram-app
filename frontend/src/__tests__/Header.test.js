import { render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";
import Header from "../components/Header/Header";

describe("Header", () => {
  it("renders the header", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the text logo", () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(screen.getAllByText("travelGram")).toBeTruthy();
  });

  it("renders the 'our users' link", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    const ourUsersLink = getByTestId("our-users-link");
    expect(ourUsersLink).toBeTruthy();
  });

  it("renders the login link", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    const login = getByTestId("login-link");
    expect(login).toBeTruthy();
  });
});
