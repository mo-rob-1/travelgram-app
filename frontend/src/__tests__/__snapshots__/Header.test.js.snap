// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header 1`] = `
<DocumentFragment>
  <header
    class="sc-bczRLJ fNAhah"
  >
    <div
      class="sc-hKMtZM bkSyjT"
    >
      <a
        class="sc-dkzDqf iflLIM"
        href="/"
      >
        <span
          class="sc-ftvSup dybfDo"
        >
          travel
        </span>
        Gram
      </a>
    </div>
    <div
      class="sc-jSMfEi heNKbs"
    >
      <nav
        class="sc-eCYdqJ jWlGGJ"
      >
        <ul
          class="sc-gsnTZi fbDbaA"
        >
          <li>
            <a
              class="sc-gKXOVf frrZiV"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="sc-gKXOVf frrZiV"
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="sc-gKXOVf frrZiV"
              data-testid="our-users-link"
              href="/our-users"
            >
              Our Users
            </a>
          </li>
          <li>
            <a
              class="sc-gKXOVf frrZiV"
              data-testid="login-link"
              href="/login"
            >
              Login
            </a>
          </li>
          <li>
            <a
              class="sc-gKXOVf frrZiV"
              data-testid="register-link"
              href="/register"
              style="margin-right: 0px;"
            >
              Register
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
